modules/scripts/startup-script/files/install_ansible.sh

```diff
@@ -15,7 +15,7 @@
 
 REQ_ANSIBLE_VERSION=2.11
 REQ_ANSIBLE_PIP_VERSION=4.10.0
-REQ_PIP_MINOR_VERSION=18
+REQ_PIP_MAJOR_VERSION=21
```

Comment:
18.1 was required for Debian 10 support. I am not sure this will be satisfiable in Debian 10 now. See [go/hpc-toolkit-dependency-install-design](http://goto.google.com/hpc-toolkit-dependency-install-design)

Reply:
Debian 10 comes with pip 18.1 in its standard repositories. So I believe the script was written in mind with "that's probably good enough and we don't need to run pip install --upgrade".

It's perfectly easy to upgrade to `pip >=21.3.1` on Debian 10 because it includes Python 3.7. RedHat 7 and 8 impose our hardest constraints on `pip` release because they come with Python 3.6. For that the maximum `pip` is `21.3.1`.

In practice, we _never, ever_ want `pip < 20.3` because [until 20.3 pip didn't fail when a user requests installation of conflicting packages](https://pip.pypa.io/en/latest/user_guide/#changes-to-the-pip-dependency-resolver-in-20-3-2020).

So, we could write `> 20.3` and probably be OK. But we might as well write the maximum version allowed by Python 3.6.

Reply:
In these release notes, you will also find why running `pip install` as root is sufficiently bad that the command itself tells you not to do it while you are doing it:

> This also means that, when you run a pip install command, pip only considers the packages you are installing in that command, and may break already-installed packages. It will not guarantee that your environment will be consistent all the time. If you pip install x and then pip install y, it’s possible that the version of y you get will be different than it would be if you had run pip install x y in a single command. We are considering changing this behavior (per [#7744](https://github.com/pypa/pip/issues/7744)) and would like your thoughts on what pip’s behavior should be; please answer [our survey on upgrades that create conflicts](https://docs.google.com/forms/d/e/1FAIpQLSeBkbhuIlSofXqCyhi3kGkLmtrpPOEBwr6iJA6SzHdxWKfqdA/viewform).